### DISCUSSION TITLE
useWatch becomes out of date when a new field is appended to the array

### DISCUSSION THREAD

**Discussion by samantha-wong** [react-hook-form/react-hook-form#4006] [Q&A] [answered]
Hi everyone, I'm pretty new to react hook forms and as a result apologise if this is a silly question. 
I've been trying to get this scenario working for a while now and I'm pretty sure useWatch isn't in sync when I append new fields to the field array. 

https://codesandbox.io/s/dazzling-dewdney-lee57?file=/pages/index.tsx

Thank you so much for your time 🙏

**Comment by bluebill1049** [comment, marked as answer]
share a codesandbox.

**Reply by samantha-wong** [reply]
Thanks @bluebill1049 for replying so quickly, here's the codesandbox: 
https://codesandbox.io/s/dazzling-dewdney-lee57?file=/pages/index.tsx

**Reply by bluebill1049** [reply]
you probably should supply a default value with `useWatch`?

**Reply by samantha-wong** [reply]
Sorry yes I needed to have a default value for useWatch, but the bigger problem is when we click on edit to try edit the form. Console log is showing the correct values, but the form doesn't know how to re-render the correct values. (I've updated the codesandbox)

**Reply by samantha-wong** [reply]
Hi again, I figured it out, I was setting the state to show parts of the form which made things really weird. Solved it by rendering the entire form and making the irrelevant fields hidden instead :)

However is there any reason why we can't only render parts of the field array, in this case a section of the form. Asking because the editor we're working on is a really big/complex form.

**Reply by bluebill1049** [reply]
This is mostly due to the design limitation. you want to try out with the latest v7 alpha version: `react-hook-form@7.0.0-alpha.0`

you can find out the change detail here: https://github.com/react-hook-form/react-hook-form/discussions/3714
